Tests are broken due to ID changes

### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Description

Most tests are failing on tip-of-tree:

```
Starting webdriverio...
  Keyboard navigation on Blocks
    ✔ Default workspace
    1) Selected block
    2) Down from statement block selects next block across stacks
    3) Up from statement block selects previous block
    4) Down from parent block selects first child block
    5) Up from child block selects parent block
    ✔ Right from block selects first field (210ms)
    6) Right from block selects first inline input
    7) Up from inline input selects statement block
    8) Left from first inline input selects block
    9) Right from first inline input selects second inline input
    10) Left from second inline input selects first inline input
    11) Right from last inline input selects next block
    12) Down from inline input selects next block
    13) Down from inline input selects block's child block
    14) Right from text block selects shadow block then field

  Keyboard navigation on Fields
    15) Up from first field selects block
    16) Left from first field selects block
    ✔ Right from first field selects second field (212ms)
    ✔ Left from second field selects first field (226ms)
    17) Right from second field selects next block
    18) Down from field selects next block
    19) Down from field selects block's child block

  Clipboard test
    ✔ Copy and paste while block selected (255ms)
    ✔ Cut and paste while block selected (780ms)
    ✔ Copy and paste whilst dragging block (443ms)
    ✔ Cut whilst dragging block (427ms)

  Insert test
    ✔ Insert C-shaped block with statement block selected (408ms)
```

### Reproduction steps

1. Check out 8a9e9d98372ada7228ae7bc5297bc6ba10b7b7f4.
2. Ensure that it's updated to use v12 beta.7 and has been correctly `npm install`'d and `npm ci`'d.
3. Run `npm run test`.
4. Observe the failures.


### Stack trace

```shell
There are a bunch (one for each failing test), and here's one example:


  19) Keyboard navigation on Fields
       Down from field selects block's child block:

      AssertionError: expected 'blockly-1p' to equal 'draw_emoji_1'
      + expected - actual

      -blockly-1p
      +draw_emoji_1

      at Context.<anonymous> (file:///usr/local/google/home/bhenning/opensource/blockly-keyboard-experimentation/test/webdriverio/test/dist/basic_test.js:316:18)
      at Generator.next (<anonymous>)
      at fulfilled (file:///usr/local/google/home/bhenning/opensource/blockly-keyboard-experimentation/test/webdriverio/test/dist/basic_test.js:9:58)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

### Screenshots

None.

### Browsers

_No response_

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -13,7 +13,7 @@ permissions:
 
 jobs:
   build_tip_of_tree_v12:
-    name: Build test (against tip-of-tree core v12)
+    name: Build test (against tip-of-tree core develop)
     runs-on: ubuntu-latest
     steps:
       - name: Checkout experimentation plugin
@@ -25,7 +25,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           repository: 'google/blockly'
-          ref: 'rc/v12.0.0'
+          ref: 'develop'
           path: core-blockly
 
       - name: Use Node.js 20.x
@@ -42,7 +42,7 @@ jobs:
           npm install
           cd ..
 
-      - name: Link latest Blockly v12
+      - name: Link latest Blockly develop
         run: |
           cd core-blockly
           npm run package
@@ -58,7 +58,7 @@ jobs:
           npm run build
 
   build:
-    name: Build test (against pinned v12)
+    name: Build test (against pinned v12.0.0)
     # Don't run pinned version checks for PRs.
     if: ${{ !github.base_ref }}
     runs-on: ubuntu-latest

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -58,7 +58,7 @@ jobs:
           npm run build
 
   build:
-    name: Build test (against pinned v12.0.0)
+    name: Build test (against pinned v12)
     # Don't run pinned version checks for PRs.
     if: ${{ !github.base_ref }}
     runs-on: ubuntu-latest

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -66,7 +66,7 @@ jobs:
           npm run test
 
   webdriverio_tests:
-    name: WebdriverIO tests (against pinned v12.0.0)
+    name: WebdriverIO tests (against pinned v12)
     # Don't run pinned version checks for PRs.
     if: ${{ !github.base_ref }}
     timeout-minutes: 10

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -14,7 +14,7 @@ permissions:
 
 jobs:
   webdriverio_tests_tip_of_tree_v12:
-    name: WebdriverIO tests (against tip-of-tree core v12)
+    name: WebdriverIO tests (against tip-of-tree core develop)
     timeout-minutes: 10
     runs-on: ${{ matrix.os }}
 
@@ -33,7 +33,7 @@ jobs:
         uses: actions/checkout@v4
         with:
           repository: 'google/blockly'
-          ref: 'rc/v12.0.0'
+          ref: 'develop'
           path: core-blockly
 
       - name: Use Node.js 20.x
@@ -50,7 +50,7 @@ jobs:
           npm install
           cd ..
 
-      - name: Link latest Blockly v12
+      - name: Link latest Blockly develop
         run: |
           cd core-blockly
           npm run package
@@ -66,7 +66,7 @@ jobs:
           npm run test
 
   webdriverio_tests:
-    name: WebdriverIO tests (against pinned v12)
+    name: WebdriverIO tests (against pinned v12.0.0)
     # Don't run pinned version checks for PRs.
     if: ${{ !github.base_ref }}
     timeout-minutes: 10

--- a/package.json
+++ b/package.json
@@ -57,7 +57,7 @@
     "@types/p5": "^1.7.6",
     "@typescript-eslint/eslint-plugin": "^6.7.2",
     "@typescript-eslint/parser": "^6.7.2",
-    "blockly": "^12.0.0-beta.4",
+    "blockly": "^12.0.0-beta.7",
     "chai": "^5.2.0",
     "eslint": "^8.49.0",
     "eslint-config-google": "^0.14.0",
@@ -73,11 +73,11 @@
     "webdriverio": "^9.12.1"
   },
   "peerDependencies": {
-    "blockly": "^12.0.0-beta.4"
+    "blockly": "^12.0.0-beta.7"
   },
   "overrides": {
     "@blockly/field-colour": {
-      "blockly": "^12.0.0-beta.4"
+      "blockly": "^12.0.0-beta.7"
     }
   },
   "publishConfig": {

--- a/package.json
+++ b/package.json
@@ -57,7 +57,7 @@
     "@types/p5": "^1.7.6",
     "@typescript-eslint/eslint-plugin": "^6.7.2",
     "@typescript-eslint/parser": "^6.7.2",
-    "blockly": "^12.0.0-beta.7",
+    "blockly": "^12.0.0",
     "chai": "^5.2.0",
     "eslint": "^8.49.0",
     "eslint-config-google": "^0.14.0",
@@ -73,11 +73,11 @@
     "webdriverio": "^9.12.1"
   },
   "peerDependencies": {
-    "blockly": "^12.0.0-beta.7"
+    "blockly": "^12.0.0"
   },
   "overrides": {
     "@blockly/field-colour": {
-      "blockly": "^12.0.0-beta.7"
+      "blockly": "^12.0.0"
     }
   },
   "publishConfig": {

--- a/test/webdriverio/test/basic_test.ts
+++ b/test/webdriverio/test/basic_test.ts
@@ -10,7 +10,7 @@ import {
   setCurrentCursorNodeById,
   setCurrentCursorNodeByIdAndFieldName,
   getCurrentFocusNodeId,
-  getFocusedConnectionType,
+  getCurrentFocusedBlockId,
   getFocusedFieldName,
   testSetup,
   testFileLocations,
@@ -45,7 +45,7 @@ suite('Keyboard navigation on Blocks', function () {
     }
 
     chai
-      .expect(await getCurrentFocusNodeId(this.browser))
+      .expect(await getCurrentFocusedBlockId(this.browser))
       .equal('controls_if_2');
   });
 
@@ -57,7 +57,9 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.keys(Key.ArrowDown);
     await this.browser.pause(PAUSE_TIME);
 
-    chai.expect(await getCurrentFocusNodeId(this.browser)).equal('p5_draw_1');
+    chai
+      .expect(await getCurrentFocusedBlockId(this.browser))
+      .equal('p5_draw_1');
   });
 
   test('Up from statement block selects previous block', async function () {
@@ -69,7 +71,7 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.pause(PAUSE_TIME);
 
     chai
-      .expect(await getCurrentFocusNodeId(this.browser))
+      .expect(await getCurrentFocusedBlockId(this.browser))
       .equal('draw_emoji_1');
   });
 
@@ -80,7 +82,9 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.pause(PAUSE_TIME);
     await this.browser.keys(Key.ArrowDown);
     await this.browser.pause(PAUSE_TIME);
-    chai.expect(await getCurrentFocusNodeId(this.browser)).equal('p5_canvas_1');
+    chai
+      .expect(await getCurrentFocusedBlockId(this.browser))
+      .equal('p5_canvas_1');
   });
 
   test('Up from child block selects parent block', async function () {
@@ -90,7 +94,9 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.pause(PAUSE_TIME);
     await this.browser.keys(Key.ArrowUp);
     await this.browser.pause(PAUSE_TIME);
-    chai.expect(await getCurrentFocusNodeId(this.browser)).equal('p5_setup_1');
+    chai
+      .expect(await getCurrentFocusedBlockId(this.browser))
+      .equal('p5_setup_1');
   });
 
   test('Right from block selects first field', async function () {
@@ -117,7 +123,7 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.pause(PAUSE_TIME);
 
     chai.assert.equal(
-      await getCurrentFocusNodeId(this.browser),
+      await getCurrentFocusedBlockId(this.browser),
       'colour_picker_1',
     );
   });
@@ -131,7 +137,7 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.pause(PAUSE_TIME);
 
     chai.assert.equal(
-      await getCurrentFocusNodeId(this.browser),
+      await getCurrentFocusedBlockId(this.browser),
       'controls_repeat_ext_1',
     );
   });
@@ -145,7 +151,7 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.pause(PAUSE_TIME);
 
     chai.assert.equal(
-      await getCurrentFocusNodeId(this.browser),
+      await getCurrentFocusedBlockId(this.browser),
       'math_modulo_1',
     );
   });
@@ -159,7 +165,7 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.pause(PAUSE_TIME);
 
     chai.assert.equal(
-      await getCurrentFocusNodeId(this.browser),
+      await getCurrentFocusedBlockId(this.browser),
       'math_number_3',
     );
   });
@@ -173,7 +179,7 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.pause(PAUSE_TIME);
 
     chai.assert.equal(
-      await getCurrentFocusNodeId(this.browser),
+      await getCurrentFocusedBlockId(this.browser),
       'math_number_2',
     );
   });
@@ -187,7 +193,7 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.pause(PAUSE_TIME);
 
     chai
-      .expect(await getCurrentFocusNodeId(this.browser))
+      .expect(await getCurrentFocusedBlockId(this.browser))
       .equal('controls_repeat_ext_1');
   });
 
@@ -200,7 +206,7 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.pause(PAUSE_TIME);
 
     chai
-      .expect(await getCurrentFocusNodeId(this.browser))
+      .expect(await getCurrentFocusedBlockId(this.browser))
       .equal('controls_repeat_ext_1');
   });
 
@@ -213,7 +219,7 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.pause(PAUSE_TIME);
 
     chai
-      .expect(await getCurrentFocusNodeId(this.browser))
+      .expect(await getCurrentFocusedBlockId(this.browser))
       .equal('text_print_1');
   });
 
@@ -225,7 +231,7 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.keys(Key.ArrowRight);
     await this.browser.pause(PAUSE_TIME);
 
-    chai.assert.equal(await getCurrentFocusNodeId(this.browser), 'text_1');
+    chai.assert.equal(await getCurrentFocusedBlockId(this.browser), 'text_1');
 
     await this.browser.keys(Key.ArrowRight);
     await this.browser.pause(PAUSE_TIME);
@@ -238,7 +244,7 @@ suite('Keyboard navigation on Blocks', function () {
     await this.browser.pause(PAUSE_TIME);
 
     chai
-      .expect(await getCurrentFocusNodeId(this.browser))
+      .expect(await getCurrentFocusedBlockId(this.browser))
       .equal('controls_repeat_1');
   });
 });
@@ -265,7 +271,10 @@ suite('Keyboard navigation on Fields', function () {
     await this.browser.keys(Key.ArrowUp);
     await this.browser.pause(PAUSE_TIME);
 
-    chai.assert.equal(await getCurrentFocusNodeId(this.browser), 'p5_canvas_1');
+    chai.assert.equal(
+      await getCurrentFocusedBlockId(this.browser),
+      'p5_canvas_1',
+    );
   });
 
   test('Left from first field selects block', async function () {
@@ -280,7 +289,10 @@ suite('Keyboard navigation on Fields', function () {
     await this.browser.keys(Key.ArrowLeft);
     await this.browser.pause(PAUSE_TIME);
 
-    chai.assert.equal(await getCurrentFocusNodeId(this.browser), 'p5_canvas_1');
+    chai.assert.equal(
+      await getCurrentFocusedBlockId(this.browser),
+      'p5_canvas_1',
+    );
   });
 
   test('Right from first field selects second field', async function () {
@@ -333,7 +345,9 @@ suite('Keyboard navigation on Fields', function () {
     await this.browser.keys(Key.ArrowRight);
     await this.browser.pause(PAUSE_TIME);
 
-    chai.expect(await getCurrentFocusNodeId(this.browser)).equal('p5_draw_1');
+    chai
+      .expect(await getCurrentFocusedBlockId(this.browser))
+      .equal('p5_draw_1');
   });
 
   test('Down from field selects next block', async function () {
@@ -348,7 +362,9 @@ suite('Keyboard navigation on Fields', function () {
     await this.browser.keys(Key.ArrowDown);
     await this.browser.pause(PAUSE_TIME);
 
-    chai.expect(await getCurrentFocusNodeId(this.browser)).equal('p5_draw_1');
+    chai
+      .expect(await getCurrentFocusedBlockId(this.browser))
+      .equal('p5_draw_1');
   });
 
   test("Down from field selects block's child block", async function () {
@@ -364,7 +380,7 @@ suite('Keyboard navigation on Fields', function () {
     await this.browser.pause(PAUSE_TIME);
 
     chai
-      .expect(await getCurrentFocusNodeId(this.browser))
+      .expect(await getCurrentFocusedBlockId(this.browser))
       .equal('draw_emoji_1');
   });
 });

--- a/test/webdriverio/test/test_setup.ts
+++ b/test/webdriverio/test/test_setup.ts
@@ -221,7 +221,7 @@ export async function setCurrentCursorNodeByIdAndFieldName(
 }
 
 /**
- * Get the ID of the block that is currently focused.
+ * Get the ID of the node that is currently focused.
  *
  * @param browser The active WebdriverIO Browser object.
  * @returns A Promise that resolves to the ID of the current cursor node.
@@ -232,6 +232,24 @@ export async function getCurrentFocusNodeId(
   return await browser.execute(() => {
     return Blockly.getFocusManager().getFocusedNode()?.getFocusableElement()
       ?.id;
+  });
+}
+
+/**
+ * Get the ID of the block that is currently focused.
+ *
+ * @param browser The active WebdriverIO Browser object.
+ * @returns A Promise that resolves to the ID of the currently focused block.
+ */
+export async function getCurrentFocusedBlockId(
+  browser: WebdriverIO.Browser,
+): Promise<string | undefined> {
+  return await browser.execute(() => {
+    const focusedNode = Blockly.getFocusManager().getFocusedNode();
+    if (focusedNode && focusedNode instanceof Blockly.BlockSvg) {
+      return focusedNode.id;
+    }
+    return undefined;
   });
 }
 
@@ -337,65 +355,4 @@ export async function tabNavigateToWorkspace(
 export async function tabNavigateForward(browser: WebdriverIO.Browser) {
   await browser.keys(webdriverio.Key.Tab);
   await browser.pause(PAUSE_TIME);
-}
-
-/**
- * Copied from blockly browser test_setup.mjs and amended for typescript
- *
- * Find a clickable element on the block and click it.
- * We can't always use the block's SVG root because clicking will always happen
- * in the middle of the block's bounds (including children) by default, which
- * causes problems if it has holes (e.g. statement inputs). Instead, this tries
- * to get the first text field on the block. It falls back on the block's SVG root.
- *
- * @param browser The active WebdriverIO Browser object.
- * @param block The block to click, as an interactable element.
- * @param clickOptions The options to pass to webdriverio's element.click function.
- * @return A Promise that resolves when the actions are completed.
- */
-export async function clickBlock(
-  browser: WebdriverIO.Browser,
-  block: ElementWithId,
-  clickOptions: webdriverio.ClickOptions,
-) {
-  const findableId = 'clickTargetElement';
-  // In the browser context, find the element that we want and give it a findable ID.
-  await browser.execute(
-    (blockId, newElemId) => {
-      const workspaceSvg = Blockly.getMainWorkspace() as Blockly.WorkspaceSvg;
-      const block = workspaceSvg.getBlockById(blockId);
-      if (block) {
-        // Ensure the block we want to click is within the viewport.
-        workspaceSvg.scrollBoundsIntoView(
-          block.getBoundingRectangleWithoutChildren(),
-        );
-        for (const input of block.inputList) {
-          for (const field of input.fieldRow) {
-            if (field instanceof Blockly.FieldLabel) {
-              const fieldSvg = field.getSvgRoot();
-              if (fieldSvg) {
-                fieldSvg.id = newElemId;
-                return;
-              }
-            }
-          }
-        }
-        // No label field found. Fall back to the block's SVG root.
-        block.getSvgRoot().id = findableId;
-      }
-    },
-    block.id,
-    findableId,
-  );
-
-  // In the test context, get the Webdriverio Element that we've identified.
-  const elem = await browser.$(`#${findableId}`);
-
-  await elem.click(clickOptions);
-
-  // In the browser context, remove the ID.
-  await browser.execute((elemId) => {
-    const clickElem = document.getElementById(elemId);
-    clickElem?.removeAttribute('id');
-  }, findableId);
 }

